lib/util/glob-util.js

```diff
@@ -160,10 +154,25 @@ function listFilesToProcess(globPatterns, options) {
         const file = path.resolve(cwd, pattern);
 
         if (shell.test("-f", file)) {
-            addFile(fs.realpathSync(file), !shell.test("-d", file));
+            const ignoredPaths = new IgnoredPaths(options);
+
+            addFile(fs.realpathSync(file), !shell.test("-d", file), ignoredPaths);
         } else {
+
+            // regex to find .hidden or /.hidden patterns, but not ./relative or ../relative
+            const globIncludesDotfiles = /(?:(?:^\.)|(?:[\/\\]\.))[^\/\\\.].*/.test(pattern);
+
+            options.dotfiles = options.dotfiles || globIncludesDotfiles;
```

Comment:
Can you add some documentation explaining this behavior? It's still unclear to me under what circumstances a dot file will be linted and when it will not. If it's not clear to us on the team, then we definitely need this to be explicitly described somewhere for end users.